clearing an object of type ‘struct player’ with no trivial copy-assignment

**Describe the bug**

```
[172/350] Building CXX object client/CMakeFiles/freeciv21-client.dir/packhand.cpp.o
/home/louis/freeciv/freeciv21/client/packhand.cpp: In function ‘void packhand_init()’:
/home/louis/freeciv/freeciv21/client/packhand.cpp:169:66: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘struct player’ with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
  169 |   memset(invisible.placeholder, 0, sizeof(*invisible.placeholder));
      |                                                                  ^
In file included from /home/louis/freeciv/freeciv21/common/achievements.h:18,
                 from /home/louis/freeciv/freeciv21/client/packhand.cpp:31:
/home/louis/freeciv/freeciv21/common/player.h:226:8: note: ‘struct player’ declared here
  226 | struct player {
      |        ^~~~~~
```

**To Reproduce**
Steps to reproduce the behavior:
1. Build in debug mode

**Expected behavior**

No such warnings

**Platform and version (please complete the following information):**
 - OS: Ubuntu 20.04
 - Freeciv21 version: `master` as of today

--- a/client/packhand.cpp
+++ b/client/packhand.cpp
@@ -151,7 +151,7 @@ void packhand_free()
   }
 
   if (nullptr != invisible.placeholder) {
-    delete[] invisible.placeholder;
+    delete invisible.placeholder;
     invisible.placeholder = nullptr;
   }
 }
@@ -166,8 +166,7 @@ static void packhand_init()
   invisible.cities = city_list_new();
 
   // Can't use player_new() here, as it will register the player.
-  invisible.placeholder = new player[1]();
-  memset(invisible.placeholder, 0, sizeof(*invisible.placeholder));
+  invisible.placeholder = new player();
   // Set some values to prevent bugs ...
   sz_strlcpy(invisible.placeholder->name, ANON_PLAYER_NAME);
   sz_strlcpy(invisible.placeholder->username, _(ANON_USER_NAME));

--- a/common/player.cpp
+++ b/common/player.cpp
@@ -474,7 +474,7 @@ struct player *player_new(struct player_slot *pslot)
 
   // Now create the player.
   log_debug("Create player for slot %d.", player_slot_index(pslot));
-  pplayer = new player[1]();
+  pplayer = new player();
   pplayer->slot = pslot;
   pslot->player = pplayer;
   pplayer->diplstates = new const player_diplstate *[player_slot_count()]();
@@ -747,7 +747,7 @@ void player_destroy(struct player *pplayer)
     rgbcolor_destroy(pplayer->rgb);
   }
 
-  delete[] pplayer;
+  delete pplayer;
   pplayer = nullptr;
   pslot->player = nullptr;
   player_slots.used_slots--;

